helm-charts/splunk-otel-collector/templates/deployment-cluster-receiver.yaml

```diff
@@ -52,14 +61,39 @@ spec:
       tolerations:
         {{ toYaml $clusterReceiver.tolerations | nindent 8 }}
       {{- end }}
-      {{- if $clusterReceiver.affinity }}
+      {{- if or $clusterReceiver.affinity (eq (include "splunk-otel-collector.distribution" .) "eks/fargate") }}
       affinity:
-        {{- toYaml $clusterReceiver.affinity | nindent 8 }}
+        {{- $clusterReceiver.affinity | mustMergeOverwrite (fromYaml (include "splunk-otel-collector.clusterReceiverPodAntiAffinity" .)) | toYaml | nindent 8 }}
       {{- end }}
       {{- if $clusterReceiver.securityContext }}
       securityContext:
         {{ toYaml $clusterReceiver.securityContext | nindent 8 }}
       {{- end }}
+      {{- if eq (include "splunk-otel-collector.distribution" .) "eks/fargate" }}
+      initContainers:
+        - name: cluster-receiver-node-discoverer
+          image: public.ecr.aws/amazonlinux/amazonlinux:latest
+          imagePullPolicy: IfNotPresent
+          command: ["bash", "-c", "/splunk-scripts/init-eks-fargate-cluster-receiver.sh"]
+          securityContext:
+            runAsUser: 0
```

Comment:
I'm curios why does the script need to be run as root?

Reply:
removed